File: src/main/java/com/ecom/pujakriti/entity/Order.java
package com.ecom.pujakriti.entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "`Order`")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "OrderID")
    private Integer orderId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "UserID", nullable = false)
    private User user;

    @Column(name = "TotalAmount", nullable = false)
    private Double totalAmount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "AddressID", nullable = false)
    private Address address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private OrderStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "DiscountCode")
    private DiscountCode discountCode;

    @Column(name = "OrderDate", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime orderDate;

    @Column(name = "cart_id")
    private String cartId;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    private List<OrderItem> orderItems;

    @OneToOne(mappedBy = "order", cascade = CascadeType.ALL)
    private Payment payment;

    public enum OrderStatus {
        PENDING, PROCESSING, SHIPPED, DELIVERED, CANCELLED
    }
}

File: admin/src/types/Order.ts
export interface AdminOrderItem {
    orderItemId: number;
    productName: string | null;
    bundleName: string | null;
    quantity: number;
    price: number;
  }
  
  export interface AdminOrder {
    orderId: number;
    userId: number;
    userName: string;
    totalAmount: number;
    address: string;
    status: 'PENDING' | 'PROCESSING' | 'SHIPPED' | 'DELIVERED' | 'CANCELLED';
    discountCode: string | null;
    discountRate: number | null;
    orderDate: string;
    orderItems: AdminOrderItem[];
    paymentID: number | null;
  }
  
  export interface Order {
    orderId: number;
    userId: number;
    totalAmount: number;
    address: number;
    status: 'PENDING' | 'PROCESSING' | 'SHIPPED' | 'DELIVERED' | 'CANCELLED';
    discountCodeId: number | null;
    orderDate: string;
    orderItems: any[];
    paymentID: number | null;
  }